### DISCUSSION TITLE
Week8: Projekt & Grading

### DISCUSSION THREAD

**Discussion by Wildeswiesel** [Wildeswiesel/Space-Waffel#10] [General]
In dieser Woche habe ich weiter an unserem Projekt weiter gearbeitet und haben folgendes implementiert 

- Eine Login Seite (ein arbeitung in Flask_LoginMangager)
- Password Hash (einarbeitung in Werkzeug.security)
- andere nutzer können jemanden eine aufgabe ertellen welche farbig anders gestaltet sind
- CSS anpassungen

Probleme Gab es bei:

- der intergration der Datenbank um den neuen Zeilen
- haben bei erstellt namen die ID vom user statt den namen bekommen
- habe vergessen was aus der DB rauszunehmen Fehler:
 ```
                  IntegrityError
                  sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) NOT NULL constraint failed: task.person
                  [SQL: INSERT INTO task (content, date, person, done, user_id, assigned_by_id) VALUES (?, ?, ?, ?, ?, ?)]
                  [parameters: ('noah', '', None, 0, '1', 1)]
                  (Background on this error at: https://sqlalche.me/e/20/gkpj)
```
- GitHub Merge Probleme mit der DB
- Fehlende Spalte in der DB Fehler:
```
                 sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such column: task.assigned_by_id
                [SQL: SELECT task.id AS task_id, task.content AS task_content, task.date AS task_date, task.person AS task_person, task.done 
                AS task_done, task.user_id AS task_user_id, task.assigned_by_id AS task_assigned_by_id 
                FROM task 
                WHERE task.user_id = ?]
                [parameters: (1,)]
                (Background on this error at: https://sqlalche.me/e/20/e3q8)
```
- Zeit

Desweiteren haben wird die grading kreterien besprochen
Link zum Repo [hier](https://github.com/Tim10022023/ToDoListPythonProject)

**Comment by SvenSrc** [comment]
Benutzt ihr auch PHP wenn ihr eine Login Seite macht? Und wie sieht die Seite bis jetzt aus?

**Reply by Wildeswiesel** [reply]
nope wir verwenden kein PHP die Login seite wird  mittels der Bibliothek Flask_login erstell mittel diesem kann ich die daten speichern und dann beim anmelden auch vergleichen 
Die daten werden in der DB gespeichert werden aber das PAsswort wird mittels werkzeug.security gehashed damit man dieses nicht einfach rauslesen kann 
die gestaltung ist ziemlich einfach der hintergrund ist andersfarbig, die knöpfe sind andersfarbig aber der rest sieht normal aus 
![image](https://github.com/Wildeswiesel/Space-Waffel/assets/152853691/402fa91b-cc9d-4713-982c-9ed59b2bf849)
